Display Meditationes

tobiasschweizer opened the pull request:
Development of a component to display the *Meditationes*

~~requires https://github.com/dhlab-basel/Knora/pull/890~~
~~requires https://github.com/dhlab-basel/Knora-ui/pull/146~~

tobiasschweizer commented:
@SepidehAlassi This PR adds some functionality to the page template so regions with transcriptions can be handled correctly (Meditationes). 

But this should also work for pages without regions.

SepidehAlassi commented:
@tobiasschweizer I will check it soon.

tobiasschweizer commented:
ok, thx.

tobiasschweizer commented:
@SepidehAlassi Do you have some time this week to look a this PR?

SepidehAlassi commented on src/app/landing-page/landing-page.component.html:
```diff
@@ -103,6 +100,18 @@ <h2 mat-card-title>Correspondance d'Euler avec Condorcet et Turgot</h2>
         <mat-divider></mat-divider>
 
+        <section>
+            <h2 mat-card-title>Meditationes, Annotationes, Animadversiones Theologicae & Philosophicae</h2>
```
I believe this long title is not self explanatory, and does not give any information about what it is. It might be better to just write here:
**Jacob Bernoulli's Scientific Notebook (Meditationes)**
Next, we can ask Martin to write a description of Meditationes which we add with full title and Bernoulli's pic to a page user is directed to by clicking on the button. In that page, we can add a button to the list of entries.

tobiasschweizer replied:
Good point!

tobiasschweizer replied:
done in https://github.com/dhlab-basel/beol/pull/42/commits/1836cdf3dfb476ecc421bab40707806b73659942

tobiasschweizer commented:
@SepidehAlassi in https://github.com/dhlab-basel/beol/pull/42/commits/097839e6e4bd54d5efd1f12e26fcad7ebb15db4b I added some functionality to automatically add figure regions in the text. This is only used in the `transcription` template. In `MathJaxDirective`, I added an input "renderFigureRegions" that defaults to false. Only if "renderFigureRegions" the new functionality will be used. 

Yet there could still be a thing I did not think of. Could you also have a look at your letters to make sure this does not break exisiting code? Thanks!

SepidehAlassi commented:
@tobiasschweizer sure! will check now!

SepidehAlassi commented:
@tobiasschweizer I just check. It seems like nothing breaks in my letters

tobiasschweizer commented:
Does the IIIF of figure regions work for you?

tobiasschweizer commented:
attempt to test figure region rendering:

```
fit('should correctly render figure regions in the html', inject([BeolService], (beolService) => {

        spyOn(beolService, 'getRegionDimsAndFile').and.callFake(
            (regionIri) => {

                const res = new ReadResourcesSequence(
                    [new ReadResource('http://rdfh.ch/0801/1111', 'http://api.knora.org/v2/Region', 'test', [], [], [], [], {})],
                    1
                );

                return of(res);

            }
        );

        expect(beolService.getRegionDimsAndFile).toHaveBeenCalledWith('');

        component.text = "<p><a href='http://rdfh.ch/0801/1111'></p>";

        component.renderFigureRegions = true;

        fixture.detectChanges();

        const compDe = fixture.debugElement;

        const spanDe: DebugElement = compDe.query(By.css('span'));

        console.log(spanDe.nativeElement.innerHTML);

    }));



});

@Component({
    template: `
        <span [mathJax]="text" [bindEvents]="bindEvents" [renderMath]="renderMath" [renderFigureRegions]="renderFigureRegions"></span>
    `
})

class TestComponent implements OnInit {

    text: string;
    bindEvents = false;
    renderMath = false;
    renderFigureRegions = false;

    ngOnInit() {
        this.text = '(\\frac{1}{2})';
    }
}
```

SepidehAlassi commented:
@tobiasschweizer yes, it works. 

<img width="268" alt="Screenshot 2019-05-24 12 17 12" src="https://user-images.githubusercontent.com/8567642/58321296-105a9000-7e1e-11e9-8448-90f5fe909354.png">

SepidehAlassi commented:
@tobiasschweizer I can not find this test. where have you implemented it?

tobiasschweizer commented:
> I can not find this test. where have you implemented it?

have not done it yet ... ;-) It was meant as a note to myself

SepidehAlassi commented:
@tobiasschweizer what was it that you asked my review for?

tobiasschweizer commented:
I have just adapted the title of the Meditationes according to your suggestion.
